encoder.go

```diff
@@ -206,12 +206,17 @@ func (d *textDecoder) Decode(t string, value []byte) (driver.Value, error) {
 	if strings.HasPrefix(t, "Array") {
 		if len(v) > 0 && v[0] == '[' && v[len(v)-1] == ']' {
 			var items []string
-			// check that array is not empty ([])
-			if len(v) > 2 {
+			subType := t[6 : len(t)-1]
+			// check if array is string encoded (['example'])
+			if (subType == "String" || strings.HasPrefix(subType, "FixedString")) && len(v) > 4 {
+				items = strings.Split(v[2:len(v)-2], "','")
+				for i, v := range items {
+					items[i] = unescape(v)
+				}
+			} else if len(v) > 2 { // check that array is not empty ([])
```

Comment:
IMHO, this condition can be checked at first, before all other staff.
- check that array is not empty.
- check that array of strings
- other cases

Reply:
Updated to what I think you had in mind? If not, if you want to include the `if`s, am happy to add into PR.